src/logger.coffee

```diff
@@ -48,6 +50,9 @@ class Logger extends EventEmitter
 
     return nextConsole
 
+  switchTransport: (transportInstance) ->
+    @_transport = setupTransport(transportInstance)
```

Comment:
The instance should probably be used as it is with no further config made by the logger. That way it is more flexible as the interface needed is only the `logMessage()` method; configuration can be made in any way – it may not be `setLogLevelPrefixes()` etc. on every custom transport.

The transport should be configured outside `Logger` before being passed in.